src/geometry/euclidean.lean

```diff
@@ -33,3 +35,634 @@ normed_add_torsor V P
+  angle_of_vectors (-x) y = real.pi - angle_of_vectors x y :=
+by rw [←angle_of_vectors_of_neg_of_neg, neg_neg, angle_of_vectors_of_neg_right]
+
+/-- The angle between the zero vector and a vector. -/
+@[simp] lemma angle_of_vectors_zero_left (x : V) : angle_of_vectors 0 x = real.pi / 2 :=
+begin
+  unfold angle_of_vectors,
+  rw [inner_zero_left, zero_div, real.arccos_zero]
+end
+
+/-- The angle between a vector and the zero vector. -/
+@[simp] lemma angle_of_vectors_zero_right (x : V) : angle_of_vectors x 0 = real.pi / 2 :=
+begin
+  unfold angle_of_vectors,
+  rw [inner_zero_right, zero_div, real.arccos_zero]
+end
+
+/-- The angle between a nonzero vector and itself. -/
+@[simp] lemma angle_of_vectors_of_self_of_nonzero {x : V} (hx : x ≠ 0) :
+  angle_of_vectors x x = 0 :=
+begin
+  unfold angle_of_vectors,
+  rw [←inner_self_eq_norm_square, div_self (λ h, hx (inner_self_eq_zero.1 h)),
+      real.arccos_one]
+end
+
+/-- The angle between a nonzero vector and its negation. -/
+@[simp] lemma angle_of_vectors_of_neg_self_of_nonzero_right {x : V} (hx : x ≠ 0) :
+  angle_of_vectors x (-x) = real.pi :=
+by rw [angle_of_vectors_of_neg_right, angle_of_vectors_of_self_of_nonzero hx, sub_zero]
+
+/-- The angle between the negation of a nonzero vector and that
+vector. -/
+@[simp] lemma angle_of_vectors_of_neg_self_of_nonzero_left {x : V} (hx : x ≠ 0) :
+  angle_of_vectors (-x) x = real.pi :=
+by rw [angle_of_vectors_comm, angle_of_vectors_of_neg_self_of_nonzero_right hx]
+
+/-- The angle between a vector and a positive multiple of a vector. -/
+@[simp] lemma angle_of_vectors_of_pos_smul_right (x y : V) {r : ℝ} (hr : 0 < r) :
+  angle_of_vectors x (r • y) = angle_of_vectors x y :=
+begin
+  unfold angle_of_vectors,
+  rw [inner_smul_right, norm_smul, real.norm_eq_abs, abs_of_nonneg (le_of_lt hr), ←mul_assoc,
+      mul_comm _ r, mul_assoc, mul_div_mul_left' _ _ (ne_of_gt hr)]
+end
+
+/-- The angle between a positive multiple of a vector and a vector. -/
+@[simp] lemma angle_of_vectors_of_pos_smul_left (x y : V) {r : ℝ} (hr : 0 < r) :
+  angle_of_vectors (r • x) y = angle_of_vectors x y :=
+by rw [angle_of_vectors_comm, angle_of_vectors_of_pos_smul_right y x hr, angle_of_vectors_comm]
+
+/-- The angle between a vector and a negative multiple of a vector. -/
+@[simp] lemma angle_of_vectors_of_neg_smul_right (x y : V) {r : ℝ} (hr : r < 0) :
+  angle_of_vectors x (r • y) = angle_of_vectors x (-y) :=
+by rw [←neg_neg r, neg_smul, angle_of_vectors_of_neg_right,
+       angle_of_vectors_of_pos_smul_right x y (neg_pos_of_neg hr), angle_of_vectors_of_neg_right]
+
+/-- The angle between a negative multiple of a vector and a vector. -/
+@[simp] lemma angle_of_vectors_of_neg_smul_left (x y : V) {r : ℝ} (hr : r < 0) :
+  angle_of_vectors (r • x) y = angle_of_vectors (-x) y :=
+by rw [angle_of_vectors_comm, angle_of_vectors_of_neg_smul_right y x hr, angle_of_vectors_comm]
+
+/-- The cosine of the angle between two vectors. -/
+lemma cos_angle_of_vectors (x y : V) :
```

Comment:
I think it would make sense to prove this lemma directly after the definition of `angle`

Reply:
Moved up.